Add dotnet logo to contexts runtime

bruno-garcia opened the pull request:
Resolves ##23465

bruno-garcia committed:
Add dotnet logo to contexts runtime

bruno-garcia commented:
That's an angry CI

bruno-garcia commented on src/sentry/static/sentry/less/group-detail.less:
```diff
@@ -227,6 +227,10 @@
     }
 
+    &.net .context-item-icon {
+      background-image: url(../images/logos/context/dotnet.svg);
```
Oh, this isn't from platformicons?

bruno-garcia replied:
I see: https://github.com/getsentry/sentry/tree/master/src/sentry/static/sentry/images/logos/context

Should I copy the file here? 
I want to add this: https://github.com/getsentry/platformicons/blob/master/svg/dotnet.svg

matejminar replied:
That's right - I pushed it there.
It's gonna look like this:

![image](https://user-images.githubusercontent.com/9060071/107484212-689d6c80-6b82-11eb-938b-0ebee45465f9.png)

If you prefer it to be rounded like this:

![image](https://user-images.githubusercontent.com/9060071/107484437-b31ee900-6b82-11eb-9736-01f8e699df5c.png)
You might wanna add:
```suggestion
      background-image: url(../images/logos/context/dotnet.svg);
      border-radius: 50%;
```

matejminar replied:
In the future we might wanna refactor these context logos to use the ones directly in the platformicons repo - some of them are now duplicated in both places.

rodolfoBee committed:
Fix file in upload release file curl example (#23569)

matejminar committed:
feat(ui): Add dotnet to context logos

mitsuhiko committed:
feat(grouping): Introduce a grouping context to strategies (#23728)

iProgramStuff committed:
ref(workflow): Improving metrics around the workflow SLO (#23751)

dashed committed:
feat: Add backdrop option to be able to prevent modals from closing when its background is clicked (#23738)

dashed committed:
fix: Pass organization prop to the simple table chart (#23736)

Fixes JAVASCRIPT-23Q9

Zylphrex committed:
fix(dashboards): epm/eps functions should save in dashboards (#23739)

The epm/eps functions expect an interval argument or start/end to be present in
the params so it can compute the interval argument. Since dashboards are not
passing in an interval argument by default, it should provide start/end to
ensure validation is successful. This change provides some hard coded start/end
params. These params can be hard coded because it is not important what the
dates are during validation as long as they are valid.

Commit:
feat(analytics): creates an advanced analytics file (#23668)

wmak committed:
fix(alerts): Incompatible Query Alert was using the wrong link (#23761)

- Currently default adds `event.type:error` instead and transactions
  adds `event.type:default` instead

markstory committed:
fix(dashboards) Fix incorrect permissions check on dashboards (#23769)

Require permissions so that accounts without dashboards can't directly
navigate to the view.

wedamija committed:
fix(metric_alerts): Make sure we track `created_by` when creating an alert rule async (1/2) (#23763)

When we handle metric alert creation async (for slack, usually), we don't pass the request user
along as a param. So we end up not knowing which user created the metric alert.

This updates the task to accept a `user_id` param and to pass it to the serializer so that we're
able to track who created it. This is split into two parts so that we can deploy the task change
first, then the change to caller, so that we don't have any issues during deploy.

markstory committed:
ref(ts) Update MultiSelectControl to typescript and fix types (#23766)


When converting MultiSelectControl to typescript I discovered that the
types for `SelectControl` were all wrong. The default export of the
type stubs for `react-select` are not for the component, but for the
state manager. This meant that our types were all wrong. Fixing those
types required numerous generics. I had to use a few `any` values as
`forwardRef` doesn't let expose generic types. I also had to use `any`
on all the refs we take on react-select because the `StateManager`
requires multiple layers of generics and even when those generics are
filled we still need to use `any`.

Co-authored-by: Priscila Oliveira <priscilawebdev@gmail.com>

Commit:
adds missing sentry_assets (#23770)

doralchan committed:
update tooltips for perf homepage v2 (#23772)

Co-authored-by: Dora Chan <dora.lchan@gmail.com>

wedamija committed:
fix(metric_alerts): Make sure we track `created_by` when creating an alert rule async (2/2) (#23764)

When we handle metric alert creation async (for slack, usually), we don't pass the request user
along as a param. So we end up not knowing which user created the metric alert.

This updates the endpoints to pass the user id along to the task.

Fixes ISSUE-1146

doralchan committed:
update to platformicons 4.1.2 (#23771)

Co-authored-by: Dora Chan <dora.lchan@gmail.com>

evanpurkhiser committed:
ref(reports): Fresh coat of paint + hierarchy (#23757)

doralchan committed:
removing extra white bg (#23774)

Co-authored-by: Dora Chan <dora.lchan@gmail.com>

evanpurkhiser committed:
ref(js): General code hygiene cleanup in api.tsx (#23723)

Co-authored-by: Priscila Oliveira <priscilawebdev@gmail.com>

Commit:
ref(migration): Create projectcodeowners table (#23715)

Objective:
We need to make a new migration to create the sentry_projectcodeowners table. This allows us to store raw CODEOWNERS files, the code mapping id that we'll use to transform into IssueOwner syntax and the transformed IssueOwner schema. This is a prerequisite for us to import and convert CODEOWNERS into Sentry IssueOwner syntax.

armenzg committed:
ci: Remove Brew cache (#23732)

It does not seem to work as expected. We can let someone else figure this out in the future.

[ci skip]

pedrofurtado committed:
fix(ui): Broken URL to add custom tags (#23535)

Commit:
check customComponent is undefined (#23776)

markstory committed:
ref(ts) Convert ChoiceMapperField to typescript (#23747)

I added some more todos on types that are fishy.

MeredithAnya committed:
fix(slack): Dont uninstall classic bot apps (#23777)

Commit:
feat(growth): mobile prompt start tracking (#23778)

MeredithAnya committed:
ref(slack): Remove slack-v2 creds and usage (#23587)

* ref(slack): Remove slack-v2 creds and usage

* update slack requests tests

* dont need verification token

* add broken test

* mock signing secret

wedamija committed:
fix(issue_alerts): Make sure we track `created_by` when creating an issue alert async (#23780)

When we handle issue alert creation async (for slack, usually), we don't pass the request user
along as a param. So we end up not knowing which user created the alert.

This updates the task to accept a `user_id` param that is used to create the `RuleActivity` row that
tracks the rule creator, and passes the `user_id` through from the endpoint

evanpurkhiser committed:
feat(report): Link out calendar days when global-views enabled (#23753)

wedamija committed:
chore(typing): Initial test of typing and introducing a mypy config file (#23606)

* chore(typing): Initial test of typing and introducing a mypy config file

This is a test of converting a file to be entirely typed. I chose something fairly self contained,
and typed and functions imported from other modules as well.

I typed our json.loads function as `Any`. I'm open to being more specific here, although from what
I've been reading it's hard to do this well until MyPy supports recursive types.

Also added in a mypy.ini, which enables strict checking and specifies a list of files to run the
checks on. I figure that the way we'd use this is to slowly convert files over time and add them to
this list, and once and entire module is added we'd add a glob match for that module instead.

Looking for feedback on this approach. I think that we can fairly easily have this run in CI using
the config file.

wedamija committed:
fix(tests): Fix flaky test_assigned_me_or_none test (#23787)

This test is flaky because one of the events ends up not being created before the current
time. Just making sure all events are created a few mins beforehand

matejminar committed:
feat(ui): Sort teams alphabetically on project detail page (#23792)

Sort teams on the project detail page alphabetically.

I also moved the "Crash free sessions" chart display option to the top as it's the first chart preselected by default.

priscilawebdev committed:
ref(releases): Fix wrong commits/filechanged empty message (#23768)

armenzg committed:
build(dev): Do not restrict using latest version of pip (#23775)

Unfortunately, the latest change has not been playing nicely for me
and folks using Python 3.6

armenzg committed:
build(pyenv) - Fully configure pyenv to work (#23628)

Unfortunately, we were not adding `eval "$(pyenv init -)"` into a developer's
shell start up configuration file, thus, `pyenv` would be available as a command yet
not changing the Python version based on `.python-version`.

This also moves pyenv logic into its own script.

iProgramStuff committed:
fix/feat(workflow): Use action/method in `incident_attachment_info` if available (#23427)

ahmedetefy committed:
feat: Bump sentry version to 0.20.0 (#23796)

Zylphrex committed:
feat(perf): Add quick trace to transaction details (#23781)

This change adds the quick trace pills to the transaction details page. The
quick trace will show if there is a root event in the trace, the current event,
and if there are any ancestors or children.

HazAT committed:
Revert "feat: Bump sentry version to 0.20.0" (#23798)

This reverts commit b2d1193d1a87438fb963df117496c2161a4350f4.

matejminar committed:
feat(ui): Consolidate empty states (#23799)

armenzg committed:
build(ci): Create backend build changes file filter and include Makefile (#23797)

Makefile changes were not triggering backend changes.

This also groups other files that I would consider backend build related changes.

markstory committed:
feat(devserver) Enable getsentry to add to devserver (#23706)

Enable getsentry to add additional worker processes. Currently getting
useful data for org stats and billing data requires knowing exactly
which processes need to be run manually, or developers manually insert
data into billing tables.

With these hooks getsentry can be configured to run the required workers
and have a smoother/simpler workflow.

scttcper committed:
feat(workflow): Track referrer in inbox out events (WOR-603) (#23785)

leedongwei committed:
feat(auth): Add more details when merging accounts (#23788)

iProgramStuff committed:
fix(workflow): Adding guard around .first().delete() (#23729)

* gtfo first()

doralchan committed:
move icons to top right (#23802)

Co-authored-by: Dora Chan <dora.lchan@gmail.com>

robinrendle committed:
fix(copy): changed alerts setting copy (#23800)

scttcper committed:
fix(ui): Reset issue stream manager between queries (#23786)

Fixes an issue with items being out of order when stream manager push preserves ids that are shared between two queries.

dashed committed:
fix: Update the Dashboard docs link to avoid a redirect (#23809)

Zylphrex committed:
feat(perf): Add links to quick trace pills (#23811)

The quick trace pills should link to the corresponding event when possible and
link to a discover query otherwise. This change adds this interaction.

- The `Root` pill will link to the span details of the root transaction
- The `Ancestors` pill will link to a discover query containing all the
  transactions in the trace
- The `This Event` pill has no additional interactions
- The `1 Child` pill will link to the span details of the only direct child
- The `X Children` pill will link to a discover query containing all the child
  transactions in the trace
- The `Descendants` pill will link to a discover query containing all the
  transactions in the trace

armenzg committed:
ci(dev_env): Test development environment on Big Sur and Ubuntu (#23703)

We have users out in the wild using `brew` on Linux (see [PR](https://github.com/getsentry/sentry/pull/23002)), thus, we will serve our community by testing the development environment there as well.

I'm also adding Big Sur back into the matrix (even if it's slower to execute) in order
to make sure we don't regress it when we make changes to the shared Brewfile.

dijonkitchen committed:
fix: typo (#23602)

robinrendle committed:
fix(copy): improve settings copy (#23810)

markstory committed:
fix(ui) Normalize OptionSelector item height (#23804)

The `SettingsLayout` page sets `line-height:1` which makes
the dropdown items too small. With this change the dropdown items are
sized consistently.

markstory committed:
ref(ui) Update another deprecated select (#23806)

I converted this one to options format, so we have fewer `choices`
later.

Commit:
feat(ui): allow setting a default category for project creation (#23814)

taylangocmen committed:
feat(alerts): Add timeline view to alert rule details (#23789)

* feat(alerts): Add timeline view to alert rule details

* Fix durations on triggers

* add incident icon to timeline

* Update src/sentry/static/sentry/app/views/alerts/rules/details/timeline.tsx

Co-authored-by: Scott Cooper <scttcper@gmail.com>

* fixing margins and display

* shorten filter

* only use has redesign for metric rules

Co-authored-by: Scott Cooper <scttcper@gmail.com>

Commit:
rename event exports (#23783)

davidenwang committed:
feat(metric_alerts): Create long-lived toasts for metric alert async slack lookup #23782

When doing async slack channel lookups on alert rule creation/edit we do a synchronous 10 second lookup for the slack channel and if that times out we then switch to an async job that could last up to 3 minutes.

However nothing on the frontend properly captures the fact that work is being done, with the first Saving alert rule... toast only lasting 6 seconds.

This PR changes the toasts to be manually added and removed based on the status of the sync/async slack channel jobs.

Hopefully this will make it more clear to the user that there is work being done on the backend and hint at them to not leave the page. Additionally, the save button on the metric alert is disabled so that users won't click it multiple times out of waiting frustration.

Fixes: ISSUE-1137

scttcper committed:
feat(workflow): Track issue clicks from inbox tab (WOR-604) (#23812)

davidenwang committed:
feat(ui): Hide unnecessary suggested assignees (#23817)

If a suggested assignee is already selected it doesn't need to be shown in the suggested list for the assignee selector. This PR filters the list of suggested assignees to not include a suggestion if it is currently used as the assignee.

ahmedetefy committed:
Bump sentry version to 0.20.0 (#23822)

evanpurkhiser committed:
chore(ts): Convert routes (#23826)

matejminar committed:
fix(ui): Improve release issues empty state on mobile (#23827)

priscilawebdev committed:
fix(pii): Fix wrong Pii replacement (#23828)

priscilawebdev committed:
ref(dynamic-sampling): Add spacing between fields and move match help info to the placeholder (#23724)

dashed committed:
feat: Update Discover header to match Issues header (#23773)


Co-authored-by: Dora Chan <dora.lchan@gmail.com>

matejminar committed:
feat(ui): Remove artifacts tab from release detail page (#23832)

Artifacts have been moved to settings many 🌔moons🌔 ago, we can get rid of the notice.

matejminar committed:
feat(ui): Remove grouping settings moved alert (#23831)

Grouping settings have been moved ~3 months ago, we should be safe to remove the notice in the spirit of keeping things clean.

markstory committed:
ref(ui) Move chart title components (#23815)

I would like to use these in getsentry for the new subscription charts.
Importing from the views directories feels like bad hygiene. Moving
these components also lets us have fewer cross view imports as well.

dashed committed:
fix: Enable backdrop option for Discover column editor modal.  (#23813)

Commit:
prevent modifying external id that is used in cloudformation (#23816)

markstory committed:
fix(ui) Fix legend and title alignment (#23838)

davidenwang committed:
fix(ui): Allow ownership rules to be blank (#23825)

When trying to delete all ownership rules in settings, due to the logic in the component, the value within the input field will reset to the initial value of the ownership rule input field

This PR changes the component to allow the text field to be empty without "resetting" to its initial value.

Fixes: WOR-561

Zylphrex committed:
ui(perf): Add trace link and fix descendants style (#23844)

1. Adds in a link to the trace id to a discover query for the trace
2. Changes the words descendants to just ellipses because Descendants more
   strongly implies they exist which we can't guarantee yet. Will be addressed
   in the future.

wedamija committed:
fix: Fix attribute error when deleting emails (#23845)

We just missed a guard here

Fixes SENTRY-NBA

evanpurkhiser committed:
chore(ts): Convert LazyLoad (#23830)

MeredithAnya committed:
feat(codeowners): Add ProjectCodeOwners GET endpoint (#23842)

* feat(codeowners): Add ProjectCodeOwners GET endpoint

* add codeMappingId

Zylphrex committed:
feat(perf): Introduce new span view to discover (#23846)

The span view in performance received new designs in #23694 flagged behind
`trace-view-quick` and `trace-view-summary`. However the links from these pills
take us to discover for the time being and the span view in discover does not
yet have these changes. This brings in the new designs behind the same flags.

taylangocmen committed:
fix(alerts): Fix fetching fetching activity for incidents (#23849)

dashed committed:
fix: Set overflow: hidden to the Dashboard widget (#23808)

armenzg committed:
dev(vscode): Remove settings that VS Code removes automatically (#23534)

It seems that VS Code automatically determines that this setting is no longer
needed and removes it automatically.

You can learn more about it here:
https://github.com/microsoft/vscode-python/wiki/AB-Experiments

[skip ci]

evanpurkhiser committed:
chore(ts): Convert ProjectContext (#23852)

markstory commented:
:wave: Hi Bruno

mitsuhiko commented:
👋

billyvg commented:
👋

dcramer commented:
i think you forgot to add tests
